### PR TITLE
docs: add branch management

### PR DESCRIPTION
**Pull request by yichengq**
/cc @kelseyhightower @xiang90 @barakmich

### CODE REVIEW

File: Documentation/branch_management.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+The master branch is our development branch. It is where all the new features go into first.
+
+If you want to try new features, pull the master branch and play on it. But the branch is not really stable because new features may introduce bugs.
+
+Before the release of the next stable version, feature PRs will be frozen. We will focus on the testing, bug-fix and documentation for one to two weeks.
+
+### Stable branches
+
+Current stable branches are `release-0.4` and `release-2.0`.
```
Comment (xiang90) [inline Documentation/branch_management.md]: can we remove this line? Or we need to maintain it. I do not think it is necessary.

File: Documentation/branch_management.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+## Branch Managemnt
```
Comment (xiang90) [inline Documentation/branch_management.md]: link this doc from top level readme?

**Comment by xiang90** [comment]
LGTM

File: Documentation/branch_management.md (hunk 1 above)
Comment (yichengq) [inline Documentation/branch_management.md]: All branches with prefix 'release-' are stable branches. Is this one more general?

**Comment by kelseyhightower** [comment]
LGTM

File: Documentation/branch_management.md (hunk 1 above)
Comment (xiang90) [inline Documentation/branch_management.md]: sure

File: README.md [hunk 3]
```diff
@@ -30,7 +30,10 @@ If you're considering etcd for production use, please see: [production-ready.md]
 
 The latest release and setup instructions are available at [GitHub][github-release].
 
+If you want to develop on latest etcd, clone the repo and checkout the [branch][branch] to work on.
```
Comment (kelseyhightower) [inline README.md]: All development occurs on the master branch which includes new features and bug fixes. Bug fixes should target the master branch and ported to the appropriate release branch as described in the [branch management](./Documentation/branch_management.md) guide.

**Pull request merged**